pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -569,27 +568,46 @@ func getFirstContainerWithSourceVolume(containers []corev1.Container) (string, s
 }
 
 // Delete deletes the component
-func (a Adapter) Delete(labels map[string]string) error {
-	spinner := log.Spinnerf("Deleting devfile component %s", a.ComponentName)
-	defer spinner.End(false)
+func (a Adapter) Delete(labels map[string]string, show bool) error {
 
-	componentExists, err := utils.ComponentExists(a.Client, a.ComponentName)
+	log.Infof("\nGathering information for component %s", a.ComponentName)
+	podSpinner := log.Spinnerf("Checking status for component %s", a.ComponentName)
+	defer podSpinner.End(false)
+
+	pod, err := a.Client.GetPodUsingComponentName(a.ComponentName)
 	if kerrors.IsForbidden(err) {
 		klog.V(4).Infof("Resource for %s forbidden", a.ComponentName)
 		// log the error if it failed to determine if the component exists due to insufficient RBACs
-		spinner.End(false)
+		podSpinner.End(false)
 		log.Warningf("%v", err)
 		return nil
+	} else if e, ok := err.(*kclient.NoPodFoundError); ok {
+		podSpinner.End(false)
+		log.Warningf("%v", e)
+		return nil
 	} else if err != nil {
 		return errors.Wrapf(err, "unable to determine if component %s exists", a.ComponentName)
 	}
 
-	if !componentExists {
-		spinner.End(false)
-		log.Warningf("Component %s does not exist", a.ComponentName)
-		return nil
+	podSpinner.End(true)
+
+	// if there are preStop events, execute them before deleting the deployment
+	preStopEvents := a.Devfile.Data.GetEvents().PreStop
+	if len(preStopEvents) > 0 {
+		if pod.Status.Phase != corev1.PodRunning {
+			return fmt.Errorf("unable to execute preStop events, pod for component %s is not running", a.ComponentName)
+		}
+
+		err = a.execDevfileEvent(preStopEvents, string(common.PreStop), a.ComponentName, pod.GetName(), show)
+		if err != nil {
+			return err
+		}
 	}
 
+	log.Infof("\nDeleting devfile component %s", a.ComponentName)
```

Comment:
Can we simply say `Deleting component` instead? I feel it sounds better with `Checking status for component` and `Gathering information for component`.

Reply:
updated, I also did a few more modifications like removing the component name from the spinner. I think it is redundant to always say component name in spinner if the header has that info already. So this is how the updated header and spinner looks:
```
Maysuns-MacBook-Pro:nodejs-ex maysun$ odo delete -f

Gathering information for component madridnode
 ✓  Checking status for component [125ms]

Executing preStop event commands for component madridnode
 •  Executing secondpoststart command "echo I am also a PostStart"  ...
 •  Executing mypoststart command "echo I am a PostStart"  ...
 ✓  Executing mypoststart command "echo I am a PostStart" [924ms]
 ✓  Executing secondpoststart command "echo I am also a PostStart" [962ms]
 ✓  Executing secondpoststart command "echo I am also a PostStart" [945ms]

Deleting component madridnode
 ✓  Deleting Kubernetes resources for component [132ms]
 ✓  Successfully deleted component
```